### PR TITLE
chore(trading): release workflow

### PR DESCRIPTION
**Pull request by bwallacee**
Creating a workflow so that we can kick off a tag/deploy/release notes with one click.

### CODE REVIEW

**Review by asiaznik — APPROVED** [review]

File: .github/workflows/release-console.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  E2E_TEST_WORKFLOW: ./.github/workflows/ci-trading-python-tests.yml
+
+jobs:
+  setup_release_branch:
+    runs-on: ubuntu-latest
+    outputs:
+      branch_name: ${{ steps.prepare_branch.outputs.branch_name }}
+    steps:
+      - name: Checkout code at commit hash
+        uses: actions/checkout@v3
```
Comment (ValentinTrinque) [inline .github/workflows/release-console.yml:27]: There is a newer version for checkout: v4

File: .github/workflows/release-console.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        with:
+          ref: ${{ github.event.inputs.commitHash }}
+
+      - name: Create Tag
+        run: git tag ${{ github.event.inputs.releaseVersion }} ${{ github.event.inputs.commitHash }}
+
+      - name: Push Tag
+        run: git push origin ${{ github.event.inputs.releaseVersion }}
```
Comment (ValentinTrinque) [inline .github/workflows/release-console.yml:35]: I thin you want this:

```suggestion
        run: git push origin tag ${{ github.event.inputs.releaseVersion }}
```

File: .github/workflows/release-console.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+      - name: Prepare release branch
+        id: prepare_branch
+        run: |
+          BRANCH_NAME="release/${{ github.event.inputs.environment }}/${{ github.event.inputs.releaseVersion }}"
+          echo "::set-output name=branch_name::$BRANCH_NAME"
```
Comment (ValentinTrinque) [inline .github/workflows/release-console.yml:41]: `::set-output` is deprecated.

Github recommends `echo "branch_name=$BRANCH_NAME" >> $GITHUB_STATE`

Ref: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

File: .github/workflows/release-console.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        description: 'Deployment environment (main or testnet)'
+        required: true
+        default: 'main'
+
+env:
+  E2E_TEST_WORKFLOW: ./.github/workflows/ci-trading-python-tests.yml
+
+jobs:
+  setup_release_branch:
+    runs-on: ubuntu-latest
```
Comment (ValentinTrinque) [inline .github/workflows/release-console.yml:22]: It's generally a good practice to pin the versions. For maintenance, I would choose the same version as other workflows.

File: .github/workflows/release-console.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+          git checkout -b $BRANCH_NAME origin/${{ github.event.inputs.environment }}
+          git reset --hard ${{ github.event.inputs.commitHash }}
+          git push --force --set-upstream origin $BRANCH_NAME
+        if: ${{ github.event.inputs.environment == 'testnet' }}
+
+  run_e2e_tests:
+    needs: setup_release_branch
+    uses: ${{ env.E2E_TEST_WORKFLOW }}
+    with:
+      github-sha: ${{ needs.setup_release_branch.outputs.branch_name }}
```
Comment (ValentinTrinque) [inline .github/workflows/release-console.yml:52]: Aren't we missing some steps ?

File: .github/workflows/release-console.yml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+  await_approval:
+    needs: run_e2e_tests
+    runs-on: ubuntu-latest
+    if: ${{ always() && github.event.inputs.environment == 'testnet' }}
```
Comment (ValentinTrinque) [inline .github/workflows/release-console.yml:57]: Based on your goal, I understood you want to require an approval only when the e2e tests fail. As a result, I think `always()` is not the right condition. It should be condition based on `run_e2e_tests`'s failure . 

I am not sure that's possible (or easy) to do with the current splitting of the jobs. I think you would be better off with the manual approval to be a step inside the job `run_e2e_tests`, that would only be triggered if the step running the e2e tests fails.

```yaml
run_e2e_tests:
  steps:
    - name: Run e2e tests
      id: e2e-tests
      run: ...
    - name: Requires manual approval
      if: steps.e2e-tests.outcome != 'success'
      run: ...
```

What do you think ?

Comment (bwallacee) [reply .github/workflows/release-console.yml]: I have updated now to always require the approval, that way we can choose to merge before the end of the test run etc. It gives us more control.

File: .github/workflows/release-console.yml (hunk 5 above)
Comment (bwallacee) [reply .github/workflows/release-console.yml]: I don't believe so. Testing on a forked repo this works.

**Review by MadalinaRaicu — APPROVED** [review]

**Pull request merged**